crates/terminal/src/terminal.rs

```diff
@@ -400,10 +402,37 @@ impl TerminalBuilder {
                 program,
                 args,
                 title_override,
-            } => Some(ShellParams {
-                program,
-                args: Some(args),
-                title_override,
+            } => Some({
+                #[cfg(target_os = "windows")]
+                {
+                    ShellParams {
+                        program: util::get_windows_system_shell(),
+                        args: {
+                            let mut args = args;
+                            args.insert(0, program);
+                            Some(vec![
+                                "-EncodedCommand".to_string(),
+                                base64::prelude::BASE64_STANDARD.encode(
+                                    args.join(" ")
+                                        .encode_utf16()
+                                        .chain([0])
+                                        .flat_map(u16::to_le_bytes)
+                                        .collect::<Vec<_>>()
+                                        .as_slice(),
+                                ),
+                            ])
+                        },
+                        title_override,
+                    }
+                }
+                #[cfg(not(target_os = "windows"))]
+                {
+                    ShellParams {
+                        program,
+                        args: Some(args),
+                        title_override,
+                    }
+                }
```

Comment:
This appears to be a potential fix for https://github.com/zed-industries/zed/issues/34348 which is great, but it should not be part of this PR which really only should bump version of alacritty. It's better to keep changes like this separate so that it is then easier to perform differential analysis should we introduce a new regression.